sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py

```diff
@@ -78,17 +80,21 @@ def obtain_token_by_refresh_token(self, refresh_token, scopes, **kwargs):
     def _process_response(self, response, scopes, now):
         # type: (dict, Iterable[str], int) -> AccessToken
         _raise_for_error(response)
+
+        # TokenCache.add mutates the response. In particular, it removes tokens.
+        original_response = copy.deepcopy(response)
```

Comment:
A copy of the response is definitionally not the original response :) Suggest `original_response_value` or `response_copy`